vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/spec/VssNodeSpecModel.kt

```diff
@@ -58,33 +60,52 @@ internal class VssNodeSpecModel(
 
     private val stringTypeName = String::class.asTypeName()
     private val vssNodeSetTypeName = Set::class.parameterizedBy(VssNode::class)
-    private val genericClassTypeName = KClass::class.asClassName().parameterizedBy(STAR).copy(nullable = true)
+    private val genericClassTypeName = KClass::class.asClassName().parameterizedBy(STAR)
+    private val genericClassTypeNameNullable = KClass::class.asClassName().parameterizedBy(STAR).copy(nullable = true)
 
-    private val datatypeProperty: TypeName
+    @OptIn(ExperimentalUnsignedTypes::class)
+    private val datatypeTypeName: TypeName
         get() {
             return when (datatype) {
                 "string" -> String::class.asTypeName()
```

Comment:
optional: It might make sense to sort the types e.g.:
```
String
StringArray
Int
IntArray
...
```

If you won't do it, then I will probably do it within #95 :)

Reply:
I introduced a new enum for the mapping but also made the sorting a bit more consistent. Feel free to change it up in #95.